drivers/interrupt_controller/intc_plic.c

```diff
@@ -461,31 +572,57 @@ static int cmd_get_stats(const struct shell *sh, size_t argc, char *argv[])
 	struct plic_stats stat = data->stats;
 
 	if (argc > 2) {
-		min_hit = (uint16_t)atoi(argv[2]);
+		min_hit = (uint16_t)shell_strtoul(argv[2], 10, &ret);
+		if (ret != 0) {
+			shell_error(sh, "Failed to parse %s: %d", argv[2], ret);
+			return ret;
+		}
 		shell_print(sh, "IRQ line with > %d hits:", min_hit);
 	}
 
-	shell_print(sh, "   IRQ        Hits\tISR(ARG)");
-	for (int i = 0; i < stat.irq_count_len; i++) {
-		if (stat.irq_count[i] > min_hit) {
+	shell_fprintf(sh, SHELL_NORMAL, "   IRQ");
+	for (int cpu_id = 0; cpu_id < arch_num_cpus(); cpu_id++) {
+		shell_fprintf(sh, SHELL_NORMAL, "  CPU%2d", cpu_id);
+	}
+	if (CONFIG_MP_NUM_CPUS > 1) {
+		shell_fprintf(sh, SHELL_NORMAL, "  Total");
+	}
+	shell_fprintf(sh, SHELL_NORMAL, "\tISR(ARG)\n");
+
+	for (int i = 0; i < config->nr_irqs; i++) {
+		int offset = (CONFIG_MP_NUM_CPUS > 1) ? arch_num_cpus() * config->nr_irqs : 0;
+		uint16_t total_hits = stat.irq_count[offset + i];
+
+		if (total_hits <= min_hit) {
+			continue;
+		}
+
+		shell_fprintf(sh, SHELL_NORMAL, "  %4d", i); /* IRQ number */
+		for (int cpu_id = 0; cpu_id < arch_num_cpus(); cpu_id++) {
+			uint32_t irq_index = cpu_id * config->nr_irqs + i;
+
+			shell_fprintf(sh, SHELL_NORMAL, "  %5d", stat.irq_count[irq_index]);
+		}
+		if (CONFIG_MP_NUM_CPUS > 1) {
+			shell_fprintf(sh, SHELL_NORMAL, "  %5d",
+				      stat.irq_count[arch_num_cpus() * config->nr_irqs + i]);
```

Comment:
The indeces into this array are fairly confusing but this line can be simplified slightly using a previously declared variable. 

```suggestion
				      stat.irq_count[offset + i]);
```

In the `plic_stats` structure. Is `irq_count` a pointer to an array that is 2+ dimensions in size? Or does some other dimension reflect values other than irq counts?

It might be good in a future PR to clean that up a bit.

Reply:
Reviewed #78140. I think it's a bit clearer now.  Is this an interleaved array based on hart contexts?

Reply:
For every instance of a PLIC, if `PLIC_SHELL` is enabled, there will be a `local_irq_count[NUM_CPUS][NUM_IRQS]`, which is pointed to by `stat.irq_count`.

I can probably make a (per instance) helper inline function to access this so that it is clearer